library/ssl_srv.c

```diff
@@ -3107,6 +3107,89 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
             return( ret );
         }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+        psa_key_attributes_t key_attributes;
+        mbedtls_ssl_handshake_params *handshake = ssl->handshake;
+        size_t ecdh_bits = 0;
+        uint8_t *p = ssl->out_msg + ssl->out_msglen;
+        const size_t header_size = 4; // curve_type, namedcurve, data length
+        const size_t data_length_size = 1;
+
+        MBEDTLS_SSL_DEBUG_MSG( 1, ( "Perform PSA-based ECDH computation." ) );
+
+        /* Convert EC group to PSA key type. */
+        if( ( handshake->ecdh_psa_type =
+                PSA_KEY_TYPE_ECC_KEY_PAIR( mbedtls_ecc_group_to_psa(
```

Comment:
```suggestion
                mbedtls_psa_parse_tls_ecc_group(
```

Reply:
Why this macro should be removed?
As far as I understand `mbedtls_psa_parse_tls_ecc_group()` returns curve id and we need to convert it to key type.

Reply:
Both codes does the same comversion, using `mbedtls_psa_parse_tls_ecc_group` would simplify the code at the expense of calling `mbedtls_ecp_curve_info_from_tls_id()`.

But here the ` == 0` test is irrelevant since even if `mbedtls_ecc_group_to_psa()` returns 0, the result would be `PSA_KEY_TYPE_ECC_KEY_PAIR_BASE` and not 0.

Reply:
Och sorry I didn't noticed that other function is used. I thought that only macro is removed.